pkg/pprof/server.go

```diff
@@ -0,0 +1,73 @@
+package pprof
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/pprof"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type server struct {
+	server   *http.Server
+	listener *net.Listener
+}
+
+func NewPprofServer(addr string) (server, error) {
+	if addr == "" || addr == "0" {
+		return server{}, nil
+	}
+
+	ln, err := net.Listen("tcp", addr)
+	if err != nil {
+		return server{}, fmt.Errorf("error listening on %s: %w", addr, err)
+	}
+	mux := http.NewServeMux()
+	srv := &http.Server{
+		Handler:           mux,
+		MaxHeaderBytes:    1 << 20,
+		IdleTimeout:       90 * time.Second, // matches http.DefaultTransport keep-alive timeout
+		ReadHeaderTimeout: 32 * time.Second,
+	}
+
+	mux.HandleFunc("/debug/pprof/", pprof.Index)
+	mux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
+	mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+	mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+	mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
```

Comment:
lol, I didn't realize that the code was there 🤦 
but, does ArgoCD already have support for enabling the profiler using vars?

Reply:
I mean, maybe I've added a feature that was already supported

Reply:
Yea I was just responding to someone [on Slack](https://cloud-native.slack.com/archives/C04SURUPDL2/p1707383452116839?thread_ts=1707319125.512419&cid=C04SURUPDL2) about this and found the old PR and your PR and issue.

> but, does ArgoCD already have support for enabling the profiler using vars?

it seems to be unconditionally enabled on the metrics port(?) if I'm reading correctly. that would be a [security issue](https://mmcloughlin.com/posts/your-pprof-is-showing) if so.

also that means the port isn't customizable

Reply:
Okey, we will replace that with my current approach, right? I mean, I have to remove the old code and just keep my PR

Reply:
Can't speak for CD folks, but I think that would make sense yea.

Reply:
So this is no longer a feature now, and is a security fix, customization of port, and docs... might make sense to rewrite as a fix?

Reply:
what do you mean?

Reply:
The PR title and description?

Reply:
Yes, the title in particular is used in the changelog and the prefix is typically used for cherry-picking (although CD has a bit different process for that). This might be a `fix!` since it's technically breaking?

Reply:
is the `!` required too in the title?

Reply:
In conventional commits, the `!` is an indication of a breaking change. This does remove the previously default enabled access to the `pprof` endpoints and so is technically breaking.

Reply:
So the previous support for `pprof` actually only added it to the Application Controller and Server (some of the components may not have existed at the time; I think only support for the Repo Server was missing at the time of #7533), so this PR also _adds_ it to the rest... which is technically a feature... 😖

So maybe this makes sense as a `feat!:` that is still backported due to the security issue? I'll let the CD Maintainers decide how to handle that 😅

Reply:
No problem , `feat!:`! xD